lib/services/address/index.js

```diff
@@ -55,9 +64,73 @@ AddressService.PREFIXES = {
   SPENTSMAP: new Buffer('05', 'hex') // Get the input that spends an output
 };
 
+AddressService.MEMPREFIXES = {
+  OUTPUTS: new Buffer('01', 'hex'), // Query mempool outputs by address
+  SPENTS: new Buffer('02', 'hex'), // Query mempool inputs by address
+  SPENTSMAP: new Buffer('03', 'hex') // Query mempool for the input that spends an output
+};
+
 AddressService.SPACER_MIN = new Buffer('00', 'hex');
 AddressService.SPACER_MAX = new Buffer('ff', 'hex');
 
+AddressService.prototype.start = function(callback) {
+  var self = this;
+
+  async.series([
+    function(next) {
+      // Flush any existing mempool index
+      if (fs.existsSync(self.mempoolIndexPath)) {
+        leveldown.destroy(self.mempoolIndexPath, next);
+      } else {
+        setImmediate(next);
+      }
+    },
+    function(next) {
+      if (!fs.existsSync(self.mempoolIndexPath)) {
+        mkdirp(self.mempoolIndexPath, next);
+      } else {
+        setImmediate(next);
+      }
+    },
+    function(next) {
+      self.mempoolIndex = levelup(
+        self.mempoolIndexPath,
+        {
+          db: self.levelupStore,
+          keyEncoding: 'binary',
+          valueEncoding: 'binary',
+          fillCache: false
+        },
+        next
+      );
+    }
+  ], callback);
+
+};
+
+AddressService.prototype.stop = function(callback) {
+  // TODO Keep track of ongoing db requests before shutting down
+  this.mempoolIndex.close(callback);
+};
+
+/**
+ * This function will set `this.dataPath` based on `this.node.network`.
+ * @private
+ */
+AddressService.prototype._setMempoolIndexPath = function() {
+  $.checkState(this.node.datadir, 'Node is expected to have a "datadir" property');
+  var regtest = Networks.get('regtest');
+  if (this.node.network === Networks.livenet) {
+    this.mempoolIndexPath = this.node.datadir + '/bitcore-addressmempool.db';
+  } else if (this.node.network === Networks.testnet) {
+    this.mempoolIndexPath = this.node.datadir + '/testnet3/bitcore-addressmempool.db';
```

Comment:
Since there is some shared code, it could be useful to have `this.node.networks.getBaseDataDir()` or at least a utility function on node, we can spawn an issue and add it in later, since this should also touch the `db.js` code.